test/smoke/testApps/AgentExceptionAndDependency/src/main/java/com/microsoft/ajl/simplecalc/SimpleTrackDependencyAndExceptionServlet.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.ajl.simplecalc;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import redis.clients.jedis.Jedis;
+
+/**
+ * Servlet implementation class SimpleTrackDependencyAndExceptionServlet
+ */
+@WebServlet(description = "Performs given calculation", urlPatterns = { "/trackData" })
+public class SimpleTrackDependencyAndExceptionServlet extends HttpServlet {
+    private static final long serialVersionUID = -7496476539225639976L;
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        ServletFuncs.getRrenderHtml(request, response);
+
+        try {
+            String redisHostname = System.getenv("REDIS");
```

Comment:
Why hard coding this variable "REDIS"? Where is it coming from?

Reply:
I thinks its come from "@WithDependencyContainers(@DependencyContainer(value="redis", portMapping="6379"))".